src/WireMock.Net/ResponseBuilders/Response.cs

```diff
@@ -272,6 +277,21 @@ public IResponseBuilder WithProxy(string proxyUrl, string clientX509Certificate2
         }
 
         /// <summary>
+        /// With Proxy URL.
+        /// </summary>
+        /// <param name="proxyUrl">The proxy url.</param>
+        /// <param name="clientX509Certificate2Filename">The X509Certificate2 file to use for client authentication.</param>
+        /// /// <param name="clientX509Certificate2Password">The X509Certificate2 password.</param>
+        /// <returns>A <see cref="IResponseBuilder"/>.</returns>
+        public IResponseBuilder WithProxy(string proxyUrl, string clientX509Certificate2Filename, string clientX509Certificate2Password)
+        {
+            Check.NotEmpty(clientX509Certificate2Filename, nameof(clientX509Certificate2Password));
```

Comment:
Fix this Check --> you check clientX509Certificate2Filename but use nameof(clientX509Certificate2Password)

And add a second check.